File: 1-2주차/11659.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

const input = [];
rl.on('line', line => {
  input.push(line);
}).on('close', () => {
  const [n, m] = input[0].split(' ').map(Number);
  const numbers = input[1].split(' ').map(Number);

  const arr = new Array(n + 1).fill(0);
  for (let i = 0; i < n; i++) {
    arr[i + 1] = arr[i] + numbers[i];
  }

  for (let i = 0; i < m; i++) {
    const [a, b] = input[2 + i].split(' ').map(Number);
    console.log(arr[b] - arr[a - 1]);
  }
});

File: 1-2주차/11659.py
import sys
input = sys.stdin.readline

n, m = map(int, input().split())
numbers = list(map(int, input().split()))

arr = [0] * (n + 1)
for i in range(n):
    arr[i + 1] = arr[i] + numbers[i]

for _ in range(m):
    a, b = map(int, input().split())
    print(arr[b] - arr[a - 1])
